src/nprintml/cli.py

```diff
@@ -1,20 +1,147 @@
 """nprintML command-line interface entry-point"""
 import argparse
+import itertools
+import os
+import pathlib
+import re
+import sys
+import time
 
-# import argparse_formatter  # FIXME
+import nprintml
+from nprintml.pipeline import Pipeline
+
+# ensure default steps of pipeline auto-load
+import nprintml.net.step
+
+
+ANIMALS = ('aardvark', 'bison', 'canary', 'dalmation', 'emu', 'falcon', 'gnu',
+           'hamster', 'impala', 'jellyfish', 'kiwi', 'lemur', 'manatee',
+           'nutria', 'okapi', 'porcupine', 'quetzal', 'roadrunner', 'seal',
+           'turtle', 'unicorn', 'vole', 'wombat', 'xerus', 'yak', 'zebra')
 
 
 def execute(argv=None, **parser_kwargs):
     """Execute the nprintml CLI command."""
-    parser = argparse.ArgumentParser(
-        description='DESCRIPTION GOES HERE',
+    try:
+        parser = build_parser(**parser_kwargs)
+
+        pipeline = Pipeline(parser)
+
+        args = parser.parse_args(argv)
+
+        check_output_directory(args)
+
+        for (step, results) in pipeline(args):
+            print(step, results, sep=' → ')
+
+        print('done →', pipeline.results)
```

Comment:
(I'm sure we'll do more with output/logging, and soon; but, this does the trick for now.)